2_segmenting_data/segmentation_utils.py

```diff
@@ -0,0 +1,251 @@
+# from cellpose.io import logger_setup
+from cellpose import models, core, io, utils
+from typing import Tuple
+
+import pathlib
+import pandas as pd
+import skimage.io as io
+
+import cv2
+import numpy as np
+
+import matplotlib.path as mplPath
+
+
+def get_outlines(image: np.ndarray, model_specs: dict) -> pd.DataFrame:
+    """finds outlines of objects using cellpose specs from model_specs and return pandas array with outlines of objects
+    Args:
+        image (np.ndarray): image with objects to segment
+        model_specs (dict): specifications for cellpose segmentation
+    Returns:
+        pd.DataFrame: dataframe with object outlines
+    """
+    # Empty list for the outlines data to be appended to
+    objects_data = []
+
+    # CellPose model specs based on the specs for the object
+    cellpose_model = models.Cellpose(gpu=True, model_type=model_specs["model_type"])
+    masks, flows, styles, diams = cellpose_model.eval(
+        image,
+        diameter=model_specs["diameter"],
+        channels=model_specs["channels"],
+        flow_threshold=model_specs["flow_threshold"],
+    )
+
+    # Remove the cell masks that are on the edges of the image
+    if model_specs["remove_edge_masks"]:
+        masks = utils.remove_edge_masks(masks)
+
+    # Appends the outlines for the objects to the list for each image site
+    outlines = utils.outlines_list(masks)
+    for outline in outlines:
+        object_data = {
+            "Outline": outline,
+        }
+        objects_data.append(object_data)
+
+    # Converts list of outlines data into pandas dataframe
+    objects_data = pd.DataFrame(objects_data)
+
+    return objects_data
+
+
+def get_locations(image: np.ndarray, model_specs: dict) -> pd.DataFrame:
```

Comment:
get outlines and get locations look very similar. Is there a way to combine essential functionality into a single function and use within both functions? Remember [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)!

Reply:
I think that is possible, essentially make a parameter for a single function that is for what type of segmentation data (or however it should be called) you want to get, either locations or outlines. I will test that out now!

Reply:
Done!